src/pages/PostPage.jsx

```diff
@@ -0,0 +1,151 @@
+            >
+              ← Back to all posts
+            </Link>
+          </div>
+
+          <div className="flex flex-col lg:flex-row gap-6 w-full">
+            <div className="w-full lg:w-1/2">
+              <div className="rounded-2xl overflow-hidden">
+                <img
+                  src={selectedPhoto}
+                  alt={currentPost.title}
+                  className="w-full h-auto"
+                />
+              </div>
+              {currentPost.photos?.length > 1 && (
+                <div className="grid grid-cols-3 sm:grid-cols-4 gap-2 mt-4">
+                  {currentPost.photos.map((photo, index) => (
+                    <img
+                      key={index}
+                      src={photo}
+                      alt={`${currentPost.title} - ${index + 1}`}
+                      onClick={() => setSelectedPhoto(photo)}
+                      className={`w-full h-20 object-cover rounded-lg cursor-pointer border ${
+                        photo === selectedPhoto
+                          ? "border-primary border-2"
+                          : "border-gray"
+                      }`}
+                    />
+                  ))}
+                </div>
+              )}
+            </div>
+
+            <div className="w-full lg:w-1/2 mt-8 lg:mt-0">
+              <h2 className="text-4xl font-extrabold font-montserrat text-primary mb-2">
+                {currentPost.title}
+              </h2>
+              <p className="text-base text-gray mb-6">{currentPost.category}</p>
+
+              <div className="mb-6">
+                <h3 className="text-lg font-semibold">Details:</h3>
+                <p>{currentPost.description}</p>
+              </div>
+
+              <div className="mb-6">
+                <h3 className="text-lg font-semibold">Address to meet:</h3>
+                <p>{currentPost.location}</p>
+                <div className="w-full h-52 bg-gray-light rounded-xl flex items-center justify-center text-gray text-sm mt-4">
+                  Map will be displayed here
+                </div>
+              </div>
+
+              <div className="mb-6">
+                <h3 className="text-lg font-semibold mb-2">
+                  Giver Information:
+                </h3>
+                <div className="flex items-center gap-3 mb-2">
+                  <img
+                    src={currentPost.user?.avatar || "/icons/avatar.svg"}
+                    alt={currentPost.user?.name || "User avatar"}
+                    className="w-5 h-5 rounded-full object-cover"
+                  />
+                  <span>{currentPost.user?.name || "Unknown user"}</span>
+                </div>
+
+                <div className="flex items-center gap-2">
+                  <img src="/icons/email.svg" alt="Email" className="w-5 h-5" />
+                  <span>{currentPost.user?.email || "No email provided"}</span>
+                </div>
+              </div>
+            </div>
+          </div>
```

Comment:
Seems that this entire area is the `Post` component - you could you it here.

```suggestion
          <Post />
```

Reply:
I moved that part into a separate `Post` component and now use it in both `PostPage` and `PostModal`. It makes everything feel way cleaner and easier to manage. Appreciate the tip!